wordpress-mobile/WordPress-iOS
# Removes the calls to update SafariCredentials.

WPAuth PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/552

Removes the calls to update `SafariCredentials`.

This is done because:

1. For magic links we can't assume that the login fields are a good source of information for the current account.
2. `SafariCredentials` is not being used in the current code, so it was simpler to remove the call.  The [only call to fetch those credentials](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/046c43cff2325f15c360fdec4f9ac0f03559351d/WordPressAuthenticator/Signin/LoginEmailViewController.swift#L313) is in a VC that's [not accessible](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/046c43cff2325f15c360fdec4f9ac0f03559351d/WordPressAuthenticator/Signin/LoginPrologueViewController.swift#L160) due to the `unifiedAuth` feature flag being [always on](https://github.com/wordpress-mobile/WordPress-iOS/blob/44bd4421069755881fff14bd3b6ee27a6bdb0573/WordPress/Classes/Utility/BuildInformation/FeatureFlag.swift#L35-L36).
3. `SafariCredentials` is an old mechanism to store login credentials.  Even if we go back to the old login flow there's a pretty strong chance we'll want to remove this class anyway.

## To test:

You'll need iCloud Keychain to be enabled for this test.

Setup: remove stored credentials

1. At the login screen, tap on the email field.
2. Tap on the key icon to pick the credentials to use.
3. Tap on "iCloud Keychain..." to view your credentials
4. If your credential is stored, tap on the information icon.
5. Delete the saved credential.

Testing steps:

1. Log in manually using your email and password.
2. Once you're logged in, iOS should ask if you want to store your credentials in iCloud.  Pick "Save".
3. Log out
4. When you tap on the email field, you should be offered to pick the right credentials by iOS.

## PR submission checklist:
